### PR TITLE
Add support for Razer Blade Pro (2019)

### PR DESCRIPTION
**Pull request by cawilliamson**
Add support for the Razer Blade Pro (2019 edition) laptop.

I have verified that this builds and works properly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ The below devices are fully feature supported by OpenRazer which means all avail
 | Razer Blade 15 (2018) Mercury                 |  1532:0240  |
 | Razer BlackWidow (2019)                       |  1532:0241  |
 | Razer Blade 15 (Mid 2019) Mercury             |  1532:0245  |
+| Razer Blade Pro 17 (2019)                     |  1532:0234  |
```
Comment (z3ntu) [inline README.md:53]: Please move your entry a few places up so it's sorted by the VID:PID value

File: driver/razerkbd_driver.h [hunk 2]
```diff
@@ -36,6 +36,7 @@
 #define USB_DEVICE_ID_RAZER_BLADE_2018_BASE 0x023b
 #define USB_DEVICE_ID_RAZER_BLADE_2018_MERCURY 0x0240
 #define USB_DEVICE_ID_RAZER_BLADE_MID_2019_MERCURY 0x0245
+#define USB_DEVICE_ID_RAZER_BLADE_PRO_2019 0x0234
```
Comment (z3ntu) [inline driver/razerkbd_driver.h:39]: Please move your device up after `USB_DEVICE_ID_RAZER_BLADE_2018` as well to have them sorted

File: install_files/appstream/io.github.openrazer.openrazer.metainfo.xml [hunk 3]
```diff
@@ -74,6 +74,7 @@
     <modalias>usb:v1532p0225d*</modalias>
     <modalias>usb:v1532p0226d*</modalias>
     <modalias>usb:v1532p0228d*</modalias>
+    <modalias>usb:v1532p0228e*</modalias>
```
Comment (z3ntu) [inline install_files/appstream/io.github.openrazer.openrazer.metainfo.xml:77]: This is wrong but you can just run `./scripts/generate_appstream_file.sh` to update this file

**Review by z3ntu — CHANGES_REQUESTED** [review]
In general LGTM, just a few comments.

Also does your Blade have an illuminated logo that can be switched on and off? If yes, please add support for that here.

**Comment by z3ntu** [comment]
bump

**Comment by cawilliamson** [comment]
I now returned my Razer Blade Pro unfortunately so don't have the device to hand to make any further amendments I'm afraid.

**Comment by legomind** [comment]
I'll see what I can do when I get the time.

**Comment by ArchonMegalon** [comment]
Wow, that is a shame that there is no support for the blade pro 2019. Is this basically dead now?

**Comment by z3ntu** [comment]
@legomind @ArchonMegalon Please test this pull request, so it can get merged!

Instructions are at https://github.com/openrazer/openrazer/wiki/Building-a-package . If you have any questions, don't hesitate to ask!

**Comment by ArchonMegalon** [comment]
That may be me, but - what is TARGET BRANCH in this case? 

Because https://github.com/openrazer/openrazer/tree/feature_bladepro2019 does not exist (anymore).

**Comment by z3ntu** [comment]
@ArchonMegalon It exists now :wink:

**Comment by ArchonMegalon** [comment]
Thank you. New to Razor. I just tried to install ubuntu (live testing it), but it seems I can only boot in safe graphics mode. Should i just install ubuntu in safe graphics? 

The screen in "normal" boot looks (and stays) like this:

**Comment by legomind** [comment]
How to test with Manjaro (ARCH)? I can build from AUR, but that wont have the changes in the test branch.

Nevermind, I just edited the aur build file... installing now.

**Comment by legomind** [comment]
Works great! No problems so far.

**Comment by ArchonMegalon** [comment]
So, I think it works. The function keys are working, which is the most important thing for me. I also installed the Polychromatic Controller, but I get a "Device Not Recognised" message there. But I think I did install the wrong version of it, because I added the repository to dpkg and tried to install openrazor from there (because I am a dumb linux newbie). 

Anyway, I think apart from the chroma-think everything works - which is actually really impressive for me. 

As soon as the ID is in the main branch I guess the Poly-Controller will find it as well. Thanks for the work!

PS: IF you want me to test anything else, I am of course more than willing to do so. Just probably not able to do so without detailed instruction... :/

**Comment by z3ntu** [comment]
Merged! Thanks for the testing!

**Pull request closed without merging**